src/sidebar/components/markdown-editor.js

```diff
@@ -213,13 +214,15 @@ function Toolbar({ isPreviewing, onCommand, onTogglePreview }) {
     let newFocusedElement = null;
     switch (e.key) {
       case 'ArrowLeft':
+      case 'Left': // IE11
```

Comment:
Rather than needing to do this in each component which handles this event, there is a way we can define a global hook that pre-processes events before they are passed to components.

On application startup (so, from `src/sidebar/index.js`) you would need something like the following:

```js
import { options } from 'preact/hooks';

const prevEventHook = options.event;
options.event = event => {
  if (event.type === 'keydown' || event.type === 'keyup') {
    // Normalize `event.key`
  }
  prevEventHook(event);
};
```

The advantage is that we can re-use this in other applications more easily.

Reply:
For reference, the full list of available hooks [is defined here](https://github.com/preactjs/preact/blob/3426ec572ca465fbe959692625dacdffab75e2a5/src/index.d.ts#L255).

Reply:
I'm only seeing `unmount`, `diffed` and `vnode`. `event` is undefined so when I override options.event I wind up with all events broken.

Reply:
> I'm only seeing unmount, diffed and vnode. event is undefined so when I override options.event I wind up with all events broken.

Ah, my choice of terminology confused things. What I talked about as "hooks" above are called `options` in the Preact docs. The `hooks/src/index.js` file refers to the _other_ kind of hook which is React hooks (`useState`, `useEffect` etc.)

My code example above was also wrong. [Per the docs](https://qa.hyp.is/2cCBVn4ZEeqNpI8AiXx0Jw/preactjs.com/guide/v10/options), the `event` hook/option needs to _return_ the event, rather than just pass it on. Here is a working example:

```js
// Quake key navigation support.
import { options } from 'preact';

const bindings = {
  a: 'ArrowLeft',
  w: 'ArrowUp',
  s: 'ArrowDown',
  d: 'ArrowRight',
};
const prevEventHook = options.event;
options.event = event => {
  if (event.key in bindings) {
    const key = bindings[event.key];
    Object.defineProperty(event, 'key', {
      configurable: true,
      enumerable: true,
      value: key,
    });
  }
  return prevEventHook ? prevEventHook(event) : event;
};
```

Reply:
awe, that makes more sense, This will be a lot better than trying to maintain special events everywhere

Reply:
Turns out that `defineProperty ` this will not work in IE11 for the event object. We should discuss more options tomorrow.